flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriter.java

```diff
@@ -279,7 +279,8 @@ FlinkKafkaInternalProducer<byte[], byte[]> getCurrentProducer() {
 
     void abortLingeringTransactions(
             Collection<KafkaWriterState> recoveredStates, long startCheckpointId) {
-        List<String> prefixesToAbort = Lists.newArrayList(transactionalIdPrefix);
+        List<String> prefixesToAbort = new ArrayList<>();
```

Comment:
_note_: I did not change this one to `Arrays.asList` because that method returns an immutable view while `prefixesToAbort` is updated a couple of lines below.